src/pip/_internal/baseparser.py

```diff
@@ -192,7 +192,17 @@ def _update_defaults(self, defaults):
                 continue
 
             if option.action in ('store_true', 'store_false', 'count'):
-                val = strtobool(val)
+                try:
+                    val = strtobool(val)
+                except ValueError:
+                    self.error(
+                        "#{0} is not a valid value. Valid true values"
```

Comment:
@pradyunsg as I went through the codebase, I realized that this option is not only for `require-virtualenv`. `store_true`, `store_false`, `count` all three are being used for multiple options. So if this error has to be action specific than it would be better to have a separate function which will decide the error message based on `option.action` which error to return. something like this - 

```

Reply:
I was just using `require-virtualenv` as an example. If you can substitute the name of the option being used, that would be nice, since it would provide more hint to the user what caused the error.

Having a function to get the correct message, sounds like a good idea. :)